### ISSUE TITLE
Configuration file "/home/$ME/.config/sddm-greeterrc" not writable

### ISSUE THREAD

**Issue by Dirleye** [sddm/sddm#1718]
I've been getting this popup on boot for a little while. Clicking "Ok" allows me to continue as normal and select my session, log in, etc.

`/var/lib/sddm` is owned by sddm and the .config folder there can be written to and read from.
I'm not sure what that file is, as my other machine does not have this issue or that file.
I'm also not sure why it's looking in `/home`'s .config instead of its own.

Creating sddm-greeterrc in `home/$ME/.config/` and giving it rw permissions (including transferring ownership to sddm) doesn't help, nor does deleting the file in `/var/lib/sddm/.config/`.

**Comment by Vogtinator** [comment]
That sounds like something is screwed up either with PAM but more likely with `/etc/profile` or similar setting `$HOME`, which they must not do.

**Comment by Dirleye** [comment]
It looks like it was caused by XDG_CONFIG_HOME being set in /etc/environment, which is odd as it's been set there for ages with no issues.
Setting it in a file in ~/.config/environment.d/ instead solves the issue.

**Comment by pallaswept** [comment]
Thanks @Dirleye you solved that one for me.!

Just for reference in case other people like me come here looking for a solution:

I had a file named /etc/environment.d/filename where I had been setting my vars, and it was working perfectly well.
So to solve this issue I tried to make ~/.config/environment.d/filename .... annnnd it did nothing. The vars in the file (ie the XDG_CONFIG_HOME in our case) wasn't set.

Checking it out, I came across https://www.freedesktop.org/software/systemd/man/environment.d.html which specifies that the files in these directories are required to have a .conf extension:

> Files must have the ".conf" extension

So I changed /etc/environment.d/filename to  /etc/environment.d/filename.conf and it still works, and ~/.config/environment.d/filename to ~/.config/environment.d/filename.conf ... and it works!

No more SDDM error, and my environment vars still work.

There are a bunch of other recommendations regarding the naming and other handling of these files, on that page, I recommend reading it if you're making use of this feature.

Thanks again @Dirleye, you saved me a bunch of time there. Hope I didn't bother you by @'ing you, just wanted to show some appreciation.
